### ISSUE TITLE
Training Cost

### ISSUE THREAD

**Issue by Yeez-lee** [yuezhouhu/residual-context-diffusion#2]
Dear Yuezhou,

Thanks for an interesting idea and open-sourced codes. I wonder what GPUs (type and numbers) and how much wallclock-wise hours for converting SDAR and LLaDA into the residual mode.

**Comment by yuezhouhu** [comment]
Hi Yize, thank you for your interest in our study! Training time differs a lot with different hardware. I did not time it accurately, but for a single 8*A100 node, the SFT training of 8B model for 5 epochs is around 100~150 hours. RCD training does not differ a lot from standard SFT and the time is slightly longer, probably 20% more than standard Sequential Denoising SFT.

**Comment by Yeez-lee** [comment]
Thanks for reply. Another question is whether it is possible to train for less than 5 epochs. Do you have any loss figures from wandb?

**Comment by yuezhouhu** [comment]
Yes. In the paper, we show that with 1 epoch of training, RCD shows near-optimal accuracies (Table 4).
In comparison, the SDAR official repo uses 10 epochs to fine-tune the model:
https://github.com/JetAstra/SDAR/blob/main/training/llama_factory_sdar/examples/train_full_sdar/sdar_4b/sdar_8b_math_cot_full.yaml
